docs/tutorials/basic/data.md

```diff
@@ -187,22 +211,361 @@ class SimpleIter:
+   2. Using [__`mx.img.ImageIter`__](http://mxnet.io/api/python/io.html#mxnet.image.ImageIter) implemented in Python which is very flexible to customization. It can read from .rec(`RecordIO`) files and raw image files.
+   3. Using [__`mx.io.ImageRecordIter`__](http://mxnet.io/api/python/io.html#mxnet.io.ImageRecordIter) implemented on the MXNet backend in C++. This is less flexible to customization but provides various language bindings.
+   4. Creating a Custom Iterator inheriting `mx.io.DataIter`
+
+
+First set the environment variable `MXNET_HOME` to the root of the MXNet source folder:
+
+
+```python
+# change this to your mxnet location
+MXNET_HOME = '/scratch/mxnet'
+```
+
+
+### Preprocessing Images
+Images can be preprocessed in may different ways. We list some of them below:
+- Using `mx.io.ImageRecordIter` which is fast but not very flexible. It is great for simple tasks like image recognition but won't work for more complex tasks like detection and segmentation.
+- Using `mx.recordio.unpack_img` (or `cv2.imread`, `skimage`, etc) + `numpy` is flexible but slow due to Python Global Interpreter Lock (GIL).
+- Using MXNet provided `mx.image` package. It stores images in [__`NDArray`__](http://mxnet.io/tutorials/basic/ndarray.html) format and leverages MXNet's [dependency engine](http://mxnet.io/architecture/note_engine.html) to automatically parallelize processing and circumvent GIL.
+
+Below, we demonstrrate some of the frequently used preprocessing routines provided by the `mx.image` package.
+
+Let's download sample images that we can work with.
+
+
+```python
+os.system('wget http://data.mxnet.io/data/test_images.tar.gz')
+os.system('tar -xf test_images.tar.gz')
+```
+
+#### Loading raw images
+`mx.image.imdecode` lets us first load the images. `imdecode` provides a similar interface to ``OpenCV``.
+**Note: ** You will still need ``OpenCV``(not the CV2 Python library) installed to use `mx.image.imdecode`.
+
+
+```python
+import cv2
+import time
+img = mx.image.imdecode(open('test_images/ILSVRC2012_val_00000001.JPEG').read())
+plt.imshow(img.asnumpy()); plt.show()
+```
+
+    (210.5357948407611, 'images decoded per second with mx.image')
+
+![](https://github.com/dmlc/web-data/blob/master/mxnet/doc/tutorials/basic/data/output_42_1.png)
+
+#### Image Transformations
+
+
+```python
+# resize to w x h
+tmp = mx.image.imresize(img, 100, 70)
+plt.imshow(tmp.asnumpy()); plt.show()
+```
+
+
+![](https://github.com/dmlc/web-data/blob/master/mxnet/doc/tutorials/basic/data/output_44_0.png)
+
+
+
+```python
+# crop a random w x h region from image
+tmp, coord = mx.image.random_crop(img, (150, 200))
+print(coord)
+plt.imshow(tmp.asnumpy()); plt.show()
+```
+
+    (80, 128, 150, 200)
+
+![](https://github.com/dmlc/web-data/blob/master/mxnet/doc/tutorials/basic/data/output_45_1.png)
+
+### Loading Data using Image Iterators
+
+Before we see how to read data using the two built-in Image Iterators,
```

Comment:
also mention that you can do more fancy things with mx.image api and custom iterator